### ISSUE TITLE
Adding Environments

### ISSUE THREAD

**Issue by okatse** [codeigniter4/CodeIgniter4#9242]
class ExceptionHandler  Line 78 

```php
            if (! str_contains($request->getHeaderLine('accept'), 'text/html')) {
                $data = (ENVIRONMENT === 'development' || ENVIRONMENT === 'testing')
                    ? $this->collectVars($exception, $statusCode)
                    : '';

```
If I add a new environment, this piece of code causes data trimming. Could this be modularized ?
Example :

```php
           $som_config = new \Config\App();

            if (! str_contains($request->getHeaderLine('accept'), 'text/html')) {
                $data = (in_array(ENVIRONMENT, $som_config->som_env))
                    ? $this->collectVars($exception, $statusCode)
                    : '';
```

**Comment by kenjis** [comment]
Why do you need this?

**Comment by okatse** [comment]
Clarify. Why do I need to add a new environment ?

**Comment by michalsn** [comment]
I wonder why we hardcoded these environments for non-HTML responses, while in `ExceptionHandler::determineView()` we rely (more or less) on the `display_errors` setting: https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Debug/ExceptionHandler.php#L137.

Is there a reason why we should not follow the same rule here?

**Comment by kenjis** [comment]
@okatse What environment do you add? Why not using `development`?

**Comment by kenjis** [comment]
@michalsn Yes, it seems it is inconsistent.

**Comment by michalsn** [comment]
It has already been fixed in `4.6` branch.
